### PR TITLE
Make graphlient run on Ruby 3.0

### PR DESCRIPTION
**Pull request by Burgestrand**
Upon trying graphlient on Ruby 3.0 I noticed a few issues, one of them being critical: https://github.com/ashkan18/graphlient/commit/effbf5d2d6721e7f0d66d0bcc1f9c9818e5ee871

All the other commits were made to get either CI to pass, tests to pass, or rubocop to pass. 😄

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ before_install:
 
 matrix:
   include:
-    - rvm: 2.3.6
+    - rvm: 2.7.2
```
Comment (Burgestrand) [inline .travis.yml:18]: This could arguably be 3.0… but I'm not sure I wanted to risk it :D

File: spec/graphlient/client_query_spec.rb [hunk 2]
```diff
@@ -91,7 +91,7 @@
         expect do
           client.execute(query, id: '42')
         end.to raise_error Graphlient::Errors::GraphQLError do |e|
-          expect(e.to_s).to eq 'Variable id of type Int was provided invalid value'
+          expect(e.to_s).to eq 'Variable $id of type Int was provided invalid value'
```
Comment (Burgestrand) [inline spec/graphlient/client_query_spec.rb:94]: This is due to allowing a newer graphql version, which was required for Ruby 3.0-compatibility.

File: .rubocop.yml [hunk 3]
```diff
@@ -1,3 +1,9 @@
+AllCops:
+  TargetRubyVersion: 2.3
+
+Lint/SplatKeywordArguments:
```
Comment (Burgestrand) [inline .rubocop.yml:4]: This is an old cop that has long since been removed.

File: .rubocop.yml [hunk 4]
```diff
@@ -1,3 +1,9 @@
+AllCops:
+  TargetRubyVersion: 2.3
```
Comment (Burgestrand) [inline .rubocop.yml:2]: We could definitely use a newer version here (this used to be picked up from `.ruby-version`), but that generated new cop warnings so I decided against it.

**Comment by dblock** [comment]
👍

**Comment by ashkan18** [comment]
Just cut a new release `0.5.0` 🙌

**Pull request merged**